CONTRIBUTING.md

```diff
@@ -73,226 +73,117 @@ We'll lay out three examples below (one that uses our format for messages, one t
 
-### Example #1: [Message-Related Test](https://github.com/osmosis-labs/osmosis/blob/f0270d04bd77cc5e1c23f7913118b3c2ba737e97/x/tokenfactory/keeper/admins_test.go#L122-L181) 
+### Example #1: [Message-Related Test]
 This type of test is mainly for functions that would be triggered by incoming messages (we interact directly with the message server since all other metadata is stripped from a message by the time it hits the msg_server):
 
 ```go
-func (suite *KeeperTestSuite) TestBurnDenom() {
-    var addr0bal int64
-
-    // Create a denom.
-    suite.CreateDefaultDenom()
-
-    // mint 10 default token for testAcc[0]
-    suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(suite.defaultDenom, 10)))
-    addr0bal += 10
-
-    for _, tc := range []struct {
-        desc      string
-        amount    int64
-        burnDenom string
-        admin     string
-        valid     bool
-    }{
-        {
-            desc:      "denom does not exist",
-            amount:    10,
-            burnDenom: "factory/osmo1t7egva48prqmzl59x5ngv4zx0dtrwewc9m7z44/evmos",
-            admin:     suite.TestAccs[0].String(),
-            valid:     false,
-        },
-        {
-            desc:      "burn is not by the admin",
-            amount:    10,
-            burnDenom: suite.defaultDenom,
-            admin:     suite.TestAccs[1].String(),
-            valid:     false,
-        },
-        {
-            desc:      "burn amount is bigger than minted amount",
-            amount:    1000,
-            burnDenom: suite.defaultDenom,
-            admin:     suite.TestAccs[1].String(),
-            valid:     false,
+func(suite *KeeperTestSuite) TestCreateDenom() {
+    testCases := map[string] struct {
+        subdenom    string
+        valid       bool
+    } {
+
+        "subdenom too long": {
+            subdenom:   "assadsadsadasdasdsadsadsadsadsadsadsklkadaskkkdasdasedskhanhassyeunganassfnlksdflksafjlkasd",
+            valid:      false,
         },
-        {
-            desc:      "success case",
-            amount:    10,
-            burnDenom: suite.defaultDenom,
-            admin:     suite.TestAccs[0].String(),
-            valid:     true,
+        "success case": {
+            subdenom: "evmos",
+			valid:    true,
         },
-    } {
-        suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
-            // Test minting to admins own account
-            _, err := suite.msgServer.Burn(sdk.WrapSDKContext(suite.Ctx), types.NewMsgBurn(tc.admin, sdk.NewInt64Coin(tc.burnDenom, 10)))
+    }
+
+    for name, tc := range testCases {
+        suite.Run(name, func() {
+            // Create a denom
+            res, err := suite.msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom(suite.TestAccs[0].String(), tc.subdenom))
```

Comment:
nit: suggest defining
```
ctx := suite.Ctx
msgServer := suite.msgServer
queryClient := suite.queryClient
```

I know it's generally not a big deal, especially since we only access each of ctx, msgServer and queryClient only once. However, since this is an example in `Contributing.md`, it would be ideal to be as close to the standards as possible